# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalResponse.h
# Commit: 84162d09eef8
# Full Hash: 84162d09eef858c42992fa6cc623284bdab1fd05
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-12 21:30:33
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r 5c3b5a384f2f -r 84162d09eef8 dom/fetch/InternalResponse.h
--- a/dom/fetch/InternalResponse.h	Thu Jan 12 15:09:44 2023 +0000
+++ b/dom/fetch/InternalResponse.h	Thu Jan 12 15:09:45 2023 +0000
@@ -57,6 +57,9 @@
 
   ParentToParentInternalResponse ToParentToParentInternalResponse();
 
+  ParentToChildInternalResponse ToParentToChildInternalResponse(
+      NotNull<mozilla::ipc::PBackgroundParent*> aBackgroundParent);
+
   enum CloneType {
     eCloneInputStream,
     eDontCloneInputStream,
@@ -337,6 +340,8 @@
 
   SafeRefPtr<InternalResponse> Unfiltered();
 
+  InternalResponseMetadata GetMetadata();
+
   ~InternalResponse();
 
  private:
@@ -351,8 +356,6 @@
   template <typename T>
   static SafeRefPtr<InternalResponse> FromIPCTemplate(const T& aIPCResponse);
 
-  InternalResponseMetadata GetMetadata();
-
   ResponseType mType;
   // A response has an associated url list (a list of zero or more fetch URLs).
   // Unless stated otherwise, it is the empty list. The current url is the last